src/sentry/processing/realtime_metrics/base.py

```diff
@@ -36,3 +40,78 @@ def increment_project_event_counter(self, project_id: int, timestamp: float) ->
             pipeline.set(key, 0, nx=True, px=self._counter_ttl)
             pipeline.incr(key)
             pipeline.execute()
+
+    # TODO: do these killswitch helpers belong here, or in a different class? LowPriorityQueueStore?
+    # This probably needs locking
+    def get_lpq_projects(self) -> Set[int]:
+        """
+        Fetches the list of projects that are currently using the low priority queue.
+
+        Returns a list of project IDs.
+        """
+        return set(
+            filter(
+                None, {_to_int(project_id) for project_id in self.inner.smembers(LPQ_MEMBERS_KEY)}
+            )
+        )
+
+    def add_project_to_lpq(self, project_id: int) -> None:
+        """
+        Moves projects to the low priority queue.
+
+        This forces all symbolication events triggered by the specified projects to be redirected to
+        the low priority queue, unless these projects have been manually excluded from the low
+        priority queue via the store.symbolicate-event-lpq-never kill switch.
+
+        Raises ``LowPriorityQueueMembershipError`` if this fails to add the specified project to the
+        low priority queue.
+        """
+
+        added = self.inner.sadd(LPQ_MEMBERS_KEY, project_id)
+
+        # Looks like this might already be in the LPQ, or redis failed to add it. This is only a
+        # problem if redis failed to add it.
+        if added == 0 and not self.inner.sismember(LPQ_MEMBERS_KEY, project_id):
+            raise LowPriorityQueueMembershipError(
+                f"Failed to move project ID {project_id} to the low priority queue"
+            )
```

Comment:
IIUC this can only happen when the key is not a set.  That basically mean it is a programming error, so maybe we should do a `sentry_sdk.capture_message()` or so in that case directly saying that the key does not appear to be a set if it wasn't added?  That saves having to create a custom error type 😄

Reply:
so i could be reading the docs incorrectly: my interpretation of the (very sparse) docs is that an error will be thrown by `sadd` if the value at the specified key isn't a set, which means that that'll cascade upwards and hopefully just axe the task. could that be incorrect, and `redis-py`'s way of indicating this problem is to return 0?

Reply:
a quick parse through the `redis` and `redis-py` source suggests that an exception will get thrown if the value's type isn't a set in an `sadd`, so i'll simplify this code in light of that 👍